### ISSUE TITLE
Auto-generate ATT&CK coverage matrices

### ISSUE THREAD

**Issue by christophetd** [DataDog/stratus-red-team#95]
Idea: automatically generate images showing the ATT&CK Tactics (not techniques) coverage

Columns: ATT&CK Tactics
Rows: Stratus Red Team attack techniques

**Comment by tmendonca28** [comment]
Hi @christophetd. I would like to contribute to this issue. Could you please provide a bit more context/detail on the above?

**Comment by christophetd** [comment]
Thanks for reaching out! The idea is to provide a graphical representation of Stratus Red Team attack techniques.

Something along the lines of:
* One graphical representation per platform (AWS, Azure, GCP, K8s)
* One column per MITRE ATT&CK Tactic (Initial access, persistence...)
* One cell for each Stratus Red Team attack technique

Available data sources:
* Using the Go API
* Using what was introduced in #218 (run `make docs` to generate a YAML file with all the attack techniques)


The output might be a Markdown table or PNG image to include in the docs. Any thoughts?

**Comment by tmendonca28** [comment]
Thank you very much for the explanation @christophetd. I've gone through the [PR](https://github.com/DataDog/stratus-red-team/pull/218)  you sent over as well.

Here is my thought process:

- I believe the [following function](https://github.com/DataDog/stratus-red-team/blob/cbb315f97d8fe827c919f61c6cb5998a8c1d2a6d/v2/tools/index.go#L17) returns a map of the data found in the index.yaml (where the techniques and metadata are found). 
- Interact with the index at [this point](https://github.com/DataDog/stratus-red-team/blob/cbb315f97d8fe827c919f61c6cb5998a8c1d2a6d/v2/tools/generate-docs.go#L23) and create a generate-coverage-matrices.go in the tools dir.
- After doing so, I will output a markdown table (demarcating each of the platforms) in the docs folder. I am unsure at the moment how the process of generating a PNG from potentially a map will look like, but will see if it's possible. 
 
Does this sound like a viable approach?

**Comment by christophetd** [comment]
Makes a lot of sense! Definitely, starting with a Markdown output makes sense. At some point we can try to output a stylized HTML output if we need something fancier.

Looking forward to your contribution!

**Comment by christophetd** [comment]
@tmendonca28 Do you still have interest in contributing?

**Comment by tmendonca28** [comment]
Definitely! Sorry for dropping the ball on my end, was a bit of a busy period.

**Comment by christophetd** [comment]
No worries! I'll leave the issue open, feel free to pick it up whenever you feel like it :)

**Comment by tmendonca28** [comment]
Hey @christophetd , Happy New Year!
I worked on the [pull request](https://github.com/DataDog/stratus-red-team/pull/613) for this (finally) and decided to go for the stylised HTML output just so it has a similar look & feel to the SRT website.
LMKWYT.
